internal/checkers/http_const.go

```diff
@@ -0,0 +1,128 @@
+package checkers
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+// HTTPConst detects situations like
+//
+//	assert.HTTPStatusCode(t, handler, "GET", "/index", nil, 200)
+//	assert.HTTPBodyContains(t, handler, "GET", "/index", nil, "counter")
+//
+// and requires
+//
+//	assert.HTTPStatusCode(t, handler, http.MethodGet, "/index", nil, http.StatusOK)
+//	assert.HTTPBodyContains(t, handler, http.MethodGet, "/index", nil, "counter")
+type HTTPConst struct{}
+
+// NewHTTPConst constructs Regexp checker.
+func NewHTTPConst() HTTPConst  { return HTTPConst{} }
+func (HTTPConst) Name() string { return "http-const" }
+
+func (checker HTTPConst) Check(_ *analysis.Pass, call *CallMeta) *analysis.Diagnostic {
+	var suggestedFixes []analysis.SuggestedFix
+	switch call.Fn.NameFTrimmed {
+	case "HTTPBody",
+		"HTTPBodyContains",
+		"HTTPBodyNotContains",
+		"HTTPError",
+		"HTTPRedirect",
+		"HTTPSuccess":
+
+		if len(call.Args) < 2 {
+			return nil
+		}
+		if suggestedFix := newHTTPMethodReplacement(call); suggestedFix != nil {
+			suggestedFixes = append(suggestedFixes, *suggestedFix)
+		} else {
+			return nil
+		}
+	case "HTTPStatusCode":
+		if len(call.Args) < 5 {
+			return nil
+		}
+		if suggestedFix := newHTTPMethodReplacement(call); suggestedFix != nil {
+			suggestedFixes = append(suggestedFixes, *suggestedFix)
+		}
+		if suggestedFix := newHTTPStatusCodeReplacement(call); suggestedFix != nil {
+			suggestedFixes = append(suggestedFixes, *suggestedFix)
+		}
+	}
+	if len(suggestedFixes) > 0 {
+		return newHTTPConstDiagnostic(checker.Name(), call, suggestedFixes)
+	}
+	return nil
+}
+
+func newHTTPMethodReplacement(call *CallMeta) *analysis.SuggestedFix {
+	if bt, ok := safeTypedBasicLit(call.Args[1], token.STRING); ok {
```

Comment:
Please, stop doing function-wide if statement. It's harder to read and to maintain


```suggestion
bt, ok := safeTypedBasicLit(call.Args[1], token.STRING)

if !ok {
return nil
}
```

I recommend using revive that could detect this pattern. early-return is one we need I think